src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -78,7 +78,12 @@ bool IsSuppressed (int id, ICustomAttributeProvider warningOrigin, out SuppressM
 					if (IsSuppressedOnElement (id, warningOriginMember, out info))
 						return true;
 
-					warningOriginMember = warningOriginMember.DeclaringType;
+					if (warningOriginMember is MethodDefinition method && method.TryGetProperty (out var property) && property != null) {
```

Comment:
nit: the check for property not being null is unnecessary, the TryGetProperty method will only return a value if true and it won't be null.